modules/base/rendering/renderabletrailorbit.cpp

```diff
@@ -0,0 +1,450 @@
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/base/rendering/renderabletrailorbit.h>
+
+#include <openspace/documentation/verifier.h>
+#include <openspace/scene/translation.h>
+
+#include <numeric>
+
+// This class is using a VBO ring buffer + a constantly updated point as follows:
+// Structure of the array with a _resolution of 16:
+//  ---------------------------------------------------------------------------------
+//  | FF |    |    |    |    |    |    |    |    |    |    |    |    |    |    |    |
+//  ---------------------------------------------------------------------------------
+//    0     1    2    3    4    5    6    7    8    9   10   11   12   13   14   15  
+//                    <------ newer in time                                    oldest
+//
+// In the begining the floating value starts at 0; this means that array element 0 is
+// updated and uploaded to the GPU at every frame. The F+1 element is the newest fixed 
+// location and F-1 element is the oldest fixed location (including wrapping around the
+// array) with the times of _lastPointTime and _firstPointTime.
+//
+// If the time progresses forwards and abs(time - _lastPointTime) becomes big enough, the
+// oldest point is removed and a new fixed location is added. In the ring buffer this
+// would be represented as:
+//  ---------------------------------------------------------------------------------
+//  |    |    |    |    |    |    |    |    |    |    |    |    |    |    |    | FF |
+//  ---------------------------------------------------------------------------------
+//    0     1    2    3    4    5    6    7    8    9   10   11   12   13   14   15  
+//                    <------ newer in time                              oldest
+//
+// Thus making the floating point traverse backwards through the array and element 0 being
+// the newest fixed point. If the time processes backwards, the floating point moves
+// towards the upper areas of the array instead.
+// In both cases, only the values that have been changed will be uploaded to the GPU.
+//
+// For the rendering, this is achieved by using an index buffer that is twice the size of
+// the vertex buffer containing identical two sequences indexing the vertex array.
+// In this example:
+// -----------------------------------------------------------------
+// |0|0|0|0|0|0|0|0|0|0|1|1|1|1|1|1|0|0|0|0|0|0|0|0|0|0|1|1|1|1|1|1|     << Values
+// |0|1|2|3|4|5|6|7|8|9|0|1|2|3|4|5|0|1|2|3|4|5|6|7|8|9|0|1|2|3|4|5| written top to bottom
```

Comment:
Took a while for me to figure out that "top to bottom" was not referring to the data structure, but to the comment itself. Maybe decrease the size of the list from 16 to 8, to make everything fit in one line?